CharlieBackend.Business/Services/AttachmentService.cs

```diff
@@ -86,23 +87,25 @@ public async Task<Result<IList<AttachmentDto>>> GetAttachmentsListAsync(Attachme
                 return Result<IList<AttachmentDto>>.GetError(ErrorCode.ValidationError, error);
             }
 
-            long accountId = _currentUserService.AccountId;
-            UserRole userRole = _currentUserService.Role;
-
-            var result = new List<AttachmentDto>();
-
-            if (userRole == UserRole.Student)
+            switch (_currentUserService.Role)
             {
-                result = await _unitOfWork.AttachmentRepository
-                    .GetAttachmentList(accountId, null, null, accountId, request.StartDate, request.FinishDate);
-            }
-            else
-            {
-                result = await _unitOfWork.AttachmentRepository
-                    .GetAttachmentList(accountId, request.CourseID, request.GroupID, accountId, request.StartDate, request.FinishDate);
+                case UserRole.Student:
+                    request.StudentAccountID = _currentUserService.EntityId;
+                    break;
+                case UserRole.Mentor:
+                    request.MentorID = _currentUserService.EntityId;
+                    break;
+                case UserRole.Secretary:
+                case UserRole.Admin:
+                    break;
+                case UserRole.NotAssigned:
+                default:
+                    throw new InvalidDataException($"Provided role {_currentUserService.Role} is not supported");
             }
```

Comment:
Please rewrite this `switch` statement using regular `if` statements for readability reasons.

Reply:
I feel like readability is subjective to the point when I feel that the Switch statement is slightly easier to read in the context.
Additionally, this way it is a little more open for change in case it will be required to insert limitations for other users.